app/layersmodel.h

```diff
@@ -52,16 +53,20 @@ class LayersModel : public QAbstractListModel
 
     QList<QgsMapLayer*> layers() const;
 
+    int defaultLayerIndex() const;
+    void setDefaultLayerIndex(int index);
 
   signals:
     void layersChanged();
+    void defaultLayerIndexChanged();
 
   public slots:
     void reloadLayers(); //when project file changes, reload all layers, etc.
 
   private:
     QgsProject* mProject;
     QList<QgsMapLayer*> mLayers; // all layers
+    int mDefaultLayerIndex;
```

Comment:
The variable is uninitialized... Initialize it either here or in the default constructor